proxy.go

```diff
@@ -73,7 +73,7 @@ func Join(local, remote net.Conn, log logging.Logger) {
 
 	transfer := func(side string, dst, src net.Conn) {
 		log.Debug("proxing %s -> %s", src.RemoteAddr(), dst.RemoteAddr())
-
+		defer dst.Close()
```

Comment:
Thanks for the PR @AkeemMcLennon  But i am not sure about the validity of it. src and dst is closed eventually, because both src & dst is interchanged at lines 97:98. So `src` is `dst` in one case and vice versa in other case.

Leaving this for @rjeczalik